### DISCUSSION TITLE
Use existing "U+1E9E" in small caps

### DISCUSSION THREAD

**Discussion by Kybernator** [adobe-fonts/source-serif#133] [Q&A] [answered]
Hello,

German user here, thank you for _Source Serif_.

You do already support the 2008 Unicode addition of _U+1E9E/LATIN CAPITAL LETTER SHARP S._

However, in small caps, there is still the old fallback of transforming ß to SS. 
(If there is a stylistic set to change that behaviour, please do tell me. BTW, where can I find documentation of the OT-features? Couldn't discover one though I certainly tried...)

Capital Sharp S should be used by default when applying smcp to Small Sharp S - this is even more important than having the glyph available as a normal capital letter, since no word actually starts with it. You do have a beautiful design for that, please use it where it is most needed.

Users who prefer the previous behaviour can very simply change the outcome, and the Swiss, for example, would not be affected at all - they do not use Capital Sharp S, but since they also do not use Small Sharp S, the suggested behaviour for smcp does not affect them at all.

A growing number of font releases in the recent years does have the default behaviour suggested here. If for any reasons you should not make it the default behaviour, please enable the use of a corresponding stylistic set.

Regards,

Bernhard

**Comment by pauldhunt** [comment]
@frankrolf what are you thoughts on this?

**Comment by frankrolf** [comment, marked as answer]
Thanks for your note, I will respond to the individual points in context:

>  there is still the old fallback of transforming ß to SS.

This behavior is neither “old” nor “new”. It’s simply one of two possible solutions.
Personally, I would find different conversion routes (uppercase conversion to SS, small cap conversion to 1E9E) confusing.

> where can I find documentation of the OT-features? 

Source Serif does not contain any OT features beyond what can be expected from a text face. Making such a documentation page/PDF is nice, but it would repeat hundreds of other, similar such documentations.
There are a few custom stylistic sets, which are documented within the font files (for example: https://github.com/adobe-fonts/source-serif/blob/main/Roman/familyGSUB.fea#L816) – this documentation should show up in the OT menu of your application. If you are curious, I encourage you to try sites like https://wakamaifondue.com or https://fontdrop.info.

> Capital Sharp S should be used by default when applying smcp to Small Sharp S

Where is this rule spelled out?

> A growing number of font releases in the recent years does have the default behavior suggested here. 

I have checked ~ 650 fonts on my hard drive (system fonts, retail fonts, open source, etc.) and less than 10% do what is suggested here.

If you insist on using the small cap variant of 1E9E you’d start from the uppercase, and apply the `c2sc` feature (“All Small Caps”) in InDesign.

I will consider adding a “prefer Germandbls” stylistic set, which then will affect both case conversions (to uppercase/to small caps).

**Comment by Kybernator** [comment]
Thank you very much for having a look at the issue, I appreciate it. While my main question has been answered by your reply (see below, "c2sc"), I will leave some remarks for context and maybe later reference for others.

> Thanks for your note, I will respond to the individual points in context:
> 
> > there is still the old fallback of transforming ß to SS.
> 
> This behavior is neither “old” nor “new”. It’s simply one of two possible solutions. Personally, I would find different conversion routes (uppercase conversion to SS, small cap conversion to 1E9E) confusing.
> 

_New_ in the sense of _only possible since 2008, due to a lack of the necessary glyph before_.

> > where can I find documentation of the OT-features?
> 
> Source Serif does not contain any OT features beyond what can be expected from a text face. Making such a documentation page/PDF is nice, but it would repeat hundreds of other, similar such documentations. There are a few custom stylistic sets, which are documented within the font files (for example: https://github.com/adobe-fonts/source-serif/blob/main/Roman/familyGSUB.fea#L816) – this documentation should show up in the OT menu of your application. 
>
It doesn't. I am not a designer, but a mere user with LibreOffice -- this has become very good at making it possible to use OT-features in a document, but is lousy at giving information about the features, and an "acquired taste" in how they can be accessed.

> If you are curious, I encourage you to try sites like https://wakamaifondue.com or https://fontdrop.info.
> 

Thank you very much for the recommendations, will do so.

> > Capital Sharp S should be used by default when applying smcp to Small Sharp S
> 
> Where is this rule spelled out?
> 

I did not intend to refer to a rule, or - even worse -  make one up, but was trying to refer to a matter of convenience for using the glyph in a logical way. Sorry if another impression was given.

> > A growing number of font releases in the recent years does have the default behavior suggested here.
> 
> I have checked ~ 650 fonts on my hard drive (system fonts, retail fonts, open source, etc.) and less than 10% do what is suggested here.
> 

Thanks a lot for investing so much time. Out of curiosity: We are talking only about fonts which do support Capital Sharp S at all, right? Several oldish fonts I would like to use do not even have that glyph :-(

> If you insist on using the small cap variant of 1E9E you’d start from the uppercase, and apply the `c2sc` feature (“All Small Caps”) in InDesign.
> 

**This is it.** That was the answer I had been hoping for. Capital Sharp S in small caps is accessible, great!

> I will consider adding a “prefer Germandbls” stylistic set, which then will affect both case conversions (to uppercase/to small caps).
>

That would be even better, of course; not only for convenience's sake, since at least LibreOffice can require manual attention to the kerning if two different OT features are applied within one word, which the current situation can require if the word starts with a captal letter that is not intended to be converted. Thank you for considering this enhancement.

Again: Thank you  (personally, and Adobe, of course) for the font.
